src/textual/widgets/_markdown.py

```diff
@@ -966,7 +966,22 @@ async def forward(self) -> None:
 
     async def _on_markdown_link_clicked(self, message: Markdown.LinkClicked) -> None:
         message.stop()
-        await self.go(message.href)
+        if message.href.find("#") == -1:
+            await self.go(message.href)
+        else:
+            link, anchor = message.href.split("#", 1)
+            await self.go(link)
+
+            def scroll_to_anchor() -> None:
+                toc = self.table_of_contents.table_of_contents
+                assert toc is not None
+                for _, name, block_id in toc:
+                    if name.lower().replace(" ", "-") == anchor:
```

Comment:
This is making a somewhat naive assumption about how fragments are generated from headers. It wont always be as simple as converting spaces to hyphens and lower casing. Consider characters with accents for example. 

I think it makes sense to match what GitHub markdown does, and replicate that.